platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/PerformanceEvent.java

```diff
@@ -1,44 +1,103 @@
 package com.mapbox.mapboxsdk.module.telemetry;
 
+import com.google.gson.Gson;
+
+import com.google.gson.JsonObject;
+
+import com.google.gson.reflect.TypeToken;
 import com.mapbox.android.telemetry.Event;
 
 import android.os.Bundle;
 import android.os.Parcel;
 
+
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 import java.util.Locale;
 
 /**
  * Generic Performance Event that can be used for performance measurements.
  * Customer measurements can be added to the bundle.
+ *
+ * Bundle is expected to have following properties:
+ * "attributes", "counters", and "metadata" with String values.
+ *
+ * Attributes: a string representing an array of name/string value pair objects.
+ * Counters: a string representing an array of name/number value pair objects.
+ * Metadata is a string representation of a JsonObject with string values for
+ * the following properties:
+ *  "os" : "android"
+ *  "version": Android release
+ *  "screenSize": screen pixel width by screen pixel height,
+ *   "country: country name in the default device locale,
+ *   "device": device name,
+ *   "abi": device's api,
+ *   "brand": device's brand
+ *   "ram": RAM size
+ *   "gpu": device's GPU
+ *   "manufacturer": Manufacturer of the device
```

Comment:
this list can be removed as metadata can be any arbitrary JsonObject, content is already shown in example below.

Reply:
cc @fabian-guerra @julianrex  I am not sure if that got changed in iOS but in DM interactions I believe iOS had metadata limited to values : 
“metadata”: {
      “version”: “9”,
      “screenSize”: “1080x1794”,
      “country”: “US”,
      “device”: “Pixel 2”,
      “abi”: “arm64-v8a”,
      “brand”: “google”,
      “ram”: “3834167296”,
      “os”: “android”,
      “gpu”: “Qualcomm, Adreno (TM) 540, OpenGL ES 3.2 V@313.0 (GIT@7bf2852, Ie32bfa6f6f) (Date:06/03/18)“,
      “manufacturer”: “Google”
  }